### ISSUE TITLE
Exception after typing search term and press enter (remote)

### ISSUE THREAD

**Issue by huguesv** [Microsoft/RTVS#2318]
I initially typed:
```
?dev.list
```

and after a while, the help window showed up for that function.

I then pressed the Home button, but it didn't seem to work.  After some time I typed `plot` in the help window search bar and pressed enter. It almost immediately crashed VS.

From event viewer:

```
Application: devenv.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.Net.HttpListenerException
   at System.Net.HttpResponseStream.BeginWrite(Byte[], Int32, Int32, System.AsyncCallback, System.Object)
   at System.IO.Stream+<>c.<BeginEndWriteAsync>b__53_0(System.IO.Stream, ReadWriteParameters, System.AsyncCallback, System.Object)
   at System.Threading.Tasks.TaskFactory`1[[System.Threading.Tasks.VoidTaskResult, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].FromAsyncTrim[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089],[System.IO.Stream+ReadWriteParameters, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]](System.__Canon, ReadWriteParameters, System.Func`5<System.__Canon,ReadWriteParameters,System.AsyncCallback,System.Object,System.IAsyncResult>, System.Func`3<System.__Canon,System.IAsyncResult,System.Threading.Tasks.VoidTaskResult>)
   at System.IO.Stream.BeginEndWriteAsync(Byte[], Int32, Int32)
   at System.IO.Stream.WriteAsync(Byte[], Int32, Int32, System.Threading.CancellationToken)
   at System.IO.Stream+<CopyToAsyncInternal>d__27.MoveNext()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(System.Threading.Tasks.Task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task)
   at Microsoft.Common.Core.StreamExtensions+<CopyAndFlushAsync>d__0.MoveNext()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(System.Threading.Tasks.Task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(System.Threading.Tasks.Task)
   at Microsoft.R.Host.Client.BrokerServices.RemoteUriWebService+<GetResponseAsync>d__4.MoveNext()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(System.Threading.Tasks.Task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task)
   at Microsoft.R.Host.Client.BrokerServices.WebServer+<DoWorkAsync>d__22.MoveNext()
   at Microsoft.Common.Core.TaskExtensions+<>c.<DoNotWaitThreadContinuation>b__6_0(System.Object)
   at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context(System.Object)
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()

```

**Comment by huguesv** [comment]
It didn't repro a second time. Will let you know if I see it again.

**Comment by huguesv** [comment]
Okay, it did repro a second time.  One thing that I noticed is that the help server seemed to stop (there was a message to that effect in REPL) after I hit home.

**Comment by karthiknadig** [comment]
I was not able to repro this, please re-open if this happens again. I suspect the server failure was caused by WebException protocol error that was not handled correctly. The fix for that is already in #2347 (change #2382 ).
